--- a/pkgs/by-name/gi/gitlab/update.py
+++ b/pkgs/by-name/gi/gitlab/update.py
@@ -347,7 +347,6 @@ def update_gitlab_elasticsearch_indexer():
 def update_all(ctx, rev: str, commit: bool):
     """Update all gitlab components to the latest stable release"""
     old_data_json = _get_data_json()
-    old_container_registry_version = get_container_registry_version()

     ctx.invoke(update_data, rev=rev)

@@ -364,12 +363,8 @@ def update_all(ctx, rev: str, commit: bool):
             old_data_json["version"], new_data_json["version"], new_data_json["rev"]
         )

-    ctx.invoke(update_gitlab_container_registry)
-    if commit:
-        new_container_registry_version = get_container_registry_version()
-        commit_container_registry(
-            old_container_registry_version, new_container_registry_version
-        )
+    ctx.invoke(update_gitlab_container_registry, rev=rev, commit=commit)
+    ctx.invoke(update_gitlab_runner, rev=rev, commit=commit)


 def commit_gitlab(old_version: str, new_version: str, new_rev: str) -> None:
